README.md
Constructr CMS 3.0
==================

ConstructrCMS (<a href="http://constructr-cms.org">http://constructr-cms.org</a>) based on FatFree-Framework, Materialize CSS, MySQL/PDO, jQuery/Javascript and Passion!

That's it for now (Version 3.1 / 2018-02-15):

	- Image editing/filter Functions
	- Clear FrontendCache's
	- FrontendCache-System (automatic, file based)
	- Frontend OutputCompression (HTML, HTML5, CSS & JS)
	- Backend Cache-System
	- Page-management
	- Create, edit, re-order/order, activate/deactivate and delete pages
	- Page specific CSS- and JS-content and metadata
	- Improved Markdown (MarkdownExtra) content-management with live-preview
	- Cross-page content-clipboard
	- Create, edit, re-order/order, activate/deavtivate and delete content-elements
	- ContactForm handling via ConstructrPostmaster
	- EASY HTML-templates with ConstructrSmartTag's
	- Asset-Management (uploads / multiple Uploads) with filter and live-preview and lightbox / pagination
	- Add, edit, delete uploads
	- User-management
	- Create, edit, delete user-accounts
	- User-rights-management 
	- Activate/deactivate user-rights
	- 2-Step login
	- Limited login-attempt's
	- User-password-resetting
	- Multi-language backend (DE | EN)


Changelog.txt
CHANGELOG
- 2018-02-15 BugFixing Backend Templates
- 2016-04-06 Bugfixing ConstructrBase Class
- 2016-01-12 Jquery Update and new merged JavaScript File
- 2016-01-06 Redirect to Offset Page after deleting, oder editing a MediaFile
- 2015-10-30 PoC TreeView of pages
- 2015-10-28 Deactivate ConstructrCMS-Backend wie ROOT/index.php option
- 2015-10-27 Improved caching (Frontend and Backend)
- 2015-10-16 FeClearCache Function (Pagemanagement View)
	- ImageResize Function (Edit ImageUpload-View) 
- 2015-10-15 Various Image Functions (rename, invert, grayscale, emboss, sepia, hflip, vflip and rotate)
- 2015-10-08 Bugfix UploadsSearchr
	- Included TimHolmanns Intense Images (https://github.com/tholman/intense-images)
- 2015-10-07 Cross-Page copy and paste Content-Elements.
	- Clipboard.JS integration in Content-Views (EDIT / NEW / NEW BEFORE / NEW AFTER)
- 2015-10-06 Added Syncscroll to Content-Preview-Areas
- 2015-09-20 Backend bugfix
- 2015-09-18 Compressed and merged backend CSS- and JS-files
- 2015-09-16 Slippry Slider-Theme
- 2015-09-15 Multi-language backend (de || en)
- 2015-09-10 BugFix, Templates
- 2015-09-04 Markdown-Extra and content link-generator
- 2015-09-03 Insert more than one image with one mouseclick
- 2015-09-02 PHP 5.4 Min.
- 2015-08-31 Limited LoginAttempts
- 2015-08-28 OutputCompression for HTML, HTML5, CSS and Javascript
- 2015-08-27 UI-Improvements, updated image upload 
- 2015-08-26 Uploads Searchengine
- 2015-08-26 Markdown Improvement for Image Dimension's via Code: ![IMAGE_NAME](IMAGE_LOCATION "IMAGE_NAME" "HEIGHT" "WIDTH")
- 2015-08-25 Bugfixes, UI-improvements, New ConstructrSmartTag {{@ CONSTRUCTR_LINK( (int) PAGE_ID ) @}}
	- Creates a textlink to page with page-id "PAGE_ID"
- 2015-08-24 New ConstructrSmartTag's
	- {{@ FIRST_LEVEL_NAV @}} - A List of pages with page_level 1
	- {{@ SECOND_LEVEL_NAV @}} - A List of pages with page_level 2
	- {{@ THIRD_LEVEL_NAV @}} - A List of pages with page_level 3
	- {{@ SUBNAV_PAGE @}} - A List of subpages of actual page
	- Bugfixes and implementing Google Material Icon Set
- 2015-08-21 MIT License (MIT)
- 2015-08-21 oneCol Template
- 2015-08-20 Bugfixes, better Navigation-MagicPattern
- 2015-08-19 UI-Improvement and Bugfixes
- 2015-08-18 ConstructrPostmaster integration - Improvement
- 2015-08-18 New UserSalt with every New/Edited UserAccount - Improvement
- 2015-08-17 New UserSalt with every Login - Improvement
	- Take care of the Upgrade Instructions above - The database tables need to be updated!
- 2015-08-17 Paginated-View in Uploads - Bugfix
- 2015-08-14 Paginated-View in Uploads
- 2015-08-13 2-Step Login
- 2015-08-11 Better URL Slug
- 2015-08-07 Little Improvements and UI-Update
- 2015-08-06 Better Lightbox and MediaFilter in Uploads
- 2015-08-05 Automatic CacheRefreshing
- 2015-08-04 Multiple Uplaods with Preview / FrontendCacheSystem (File based)
- 2015-08-03 Smart Constructr Content Mapping - Map a Content-Element to a specific HTML-ID-Area / Template Update
- 2015-07-30 Split Images and other Files in InsertingModal (Content new / Content edit), minor Bugfixes and Improvements
- 2015-07-29 AutoInsert Markdown for Files / little Bugfixes and minor Features
- 2015-07-28 Webinstaller / little Bugfixes and minor Features
- 2015-07-23 Grande Update IV...
- 2015-03-23 Grande Update III...
- 2015-03-21 Grande Update II...
- 2015-03-19 Grande Update...
- 2015-03-18 Initial commit


CONSTRUCTR-CMS/LANG/en.php
<?php

	$CONSTRUCTR_LANG = [
		// NAVIGATION
		'N0' => 'Pages',
		'N1' => 'Uploads',
		'N2' => 'User',
		'N3' => ' logout?',
		'N4' => 'Logout?',
		// GENERELLES
		'G0' => 'Yes',
		'G1' => 'Cancel',
		'G2' => 'Okay',
		'G3' => 'Attention',
		'G4' => 'Ups... unknown error occured...',
		// LOGIN SEITE 1 von 2
		0 => 'Please insert your username',
		1 => 'Username',
		2 => 'eMail',
		3 => 'Login Step 1 of 2',
		4 => 'Lost password?',
		5 => 'Attention',
		6 => 'Please insert your username!',
		7 => 'Reset the password for user ',
		8 => ' ?',
		9 => 'Login',
		// LOGIN SEITE 2 von 2
		10 => 'Login',
		11 => 'Please insert your password',
		12 => 'eMail',
		13 => 'Password',
		14 => 'Login Step 2 of 2',
		// LOGIN FEHLERSEITE
		15 => 'Login-Error',
		16 => 'Please insert your username',
		17 => 'Unknown user!',
		18 => 'Username',
		19 => 'eMail',
		20 => 'Login Step 1 of 2',
		21 => 'Lost password?',
		22 => 'Attention',
		23 => 'Please insert your username!',
		24 => 'Reset the password for user ',
		25 => ' ?',
		// Passwort zurücksetzen
		26 => 'Login',
		27 => 'Login',
		28 => 'Password reset - Your new password is in your email-inbox!',
		29 => 'Username',
		30 => 'eMail',
		31 => 'Login Step 1 of 2',
		32 => 'Lost password?',
		33 => 'Attention',
		34 => 'Please insert your username!',
		35 => 'Reset the password now?',
		// GENERELLE FEHLERSEITE
		36 => 'Constructr 5** - Server error',
		37 => '5** - Server error',
		// Seitenverwaltung
		38 => 'Pagemanagement - Constructr Administration',
		39 => 'Pages',
		40 => 'New page',
		41 => 'Page edited successfully!',
		42 => 'Page not edited successfully!',
		43 => 'Visibility was not edited - this is your homepage!',
		44 => 'Page not edited successfully - URL already exists!',
		45 => 'Page not edited successfully - constructr is reserved for this administration-backend!',
		46 => 'Page created successfully!',
		47 => 'Page not created successfully!',
		48 => 'Page not created successfully - URL already exists!',
		49 => 'Page not created successfully - constructr is reserved for this administration-backend!',
		50 => 'Page deleted successfully!',
		51 => 'Page not deleted successfully!',
		52 => 'Page not deleted successfully - there is content on this page!',
		53 => 'Page not deleted successfully - it is your homepage!',
		54 => 'Page moved successfully!',
		55 => 'Page not moved successfully!',
		56 => 'Page not moved successfully - it is not allowed to overwrite the homepage!',
		57 => 'Name',
		58 => 'ID',
		59 => 'URL/frontend',
		60 => 'Template',
		61 => 'Actions',
		62 => 'Doubleclick to edit page ',
		63 => ' ',
		64 => 'Edit content on page ',
		65 => ' ',
		66 => 'Open page ',
		67 => 'Move page ',
		68 => ' upwards',
		69 => 'Move page ',
		70 => ' downwards',
		71 => 'Edit content of page ',
		72 => ' ',
		73 => 'Edit page ',
		74 => ' ',
		75 => 'Show page ',
		76 => ' in frontend-navigation',
		77 => 'Hide page ',
		78 => ' in frontend-navigation',
		79 => 'Open page ',
		80 => 'Delete page ',
		81 => ' ',
		82 => 'Delete page ',
		83 => ' and all content-elements',
		84 => 'Pages',
		85 => 'New Page',
		86 => 'Realy delete this page including all content-elements?',
		87 => 'Realy delete this page?',
		88 => 'Attention',
		// NEUE SEITE ERSTELLEN
		89 => 'New Page - Constructr Administration',
		90 => 'New Page',
		91 => 'Name of page',
		92 => 'Please select',
		93 => 'before existing page',
		94 => 'as subpage of existing page',
		95 => 'after existing page',
		96 => 'Positioning of this page',
		97 => 'Please select',
		98 => 'URL of page',
		99 => 'URL of page',
		100 => 'External forwarding (URL)',
		101 => 'Please select',
		102 => 'Template',
		103 => 'Please select',
		104 => 'Page is visible in frontend-navigation',
		105 => 'Page is hidden in frontend-navigation',
		106 => 'Visibility of this page',
		107 => 'Meta-Titel of this page',
		108 => 'Meta-Description of this page',
		109 => 'Meta-Keywords of this page',
		110 => 'Additional page CSS',
		111 => 'Additional page Javascript',
		112 => 'Save new page',
		113 => 'Cancel',
		114 => 'Press key "ESC" to return to previous view',
		// SEITE BEARBEITEN
		115 => 'Edit page - Constructr Administration',
		116 => 'Page ',
		117 => ' edit',
		118 => 'Name of page',
		119 => 'URL of page',
		120 => 'URL of page',
		121 => 'External forwarding (URL)',
		122 => 'Please select',
		123 => 'Template',
		124 => 'Please select',
		125 => 'Page is visible in frontend-navigation',
		126 => 'Page is hidden in frontend-navigation',
		127 => 'Visibility of this page',
		128 => 'Meta-Title of this page',
		129 => 'Meta-Description of this page',
		130 => 'Meta-Keywords of this page',
		131 => 'Additional page CSS',
		132 => 'Addition page Javascript',
		133 => 'Save page',
		134 => 'Cancel',
		135 => 'Press key "ESC" to return to previous view',
		// UPLOADS
		136 => 'Uploads - Constructr Administration',
		137 => 'Needle(s) seperated by space-character - hit ENTER-key for search',
		138 => 'Uploads',
		139 => 'Pictures',
		140 => 'Files',
		141 => 'New upload',
		142 => 'File(s) uploaded successfully!',
		143 => 'File(s) not uploaded successfully!',
		144 => 'File(s) deleted successfully!',
		145 => 'File(s) not deleted successfully!',
		146 => 'File/Preview',
		147 => 'Absolut URL',
		148 => 'Actions',
		149 => 'Preview',
		150 => 'Preview in new Browser-tab',
		151 => 'Delete file',
		152 => 'Realy delete the selected file?',
		// NEUER UPLOAD
		153 => 'New upload - Constructr Administration',
		154 => 'New upload (on or more files at once)',
		155 => 'Add file(s)',
		156 => 'Uplaod file(s)',
		157 => 'Cancel',
		158 => 'Press key "ESC" to return to previous view',
		// KEINE RECHTE
		159 => 'No user-right found for this action - Constructr Administration',
		160 => 'No user-right found for this action!',
		// BACKEND 404
		161 => 'Constructr 404 - not found',
		162 => '404 - Page/View not found!',
		// USERMANAGEMENT
		163 => 'User - Constructr Administration',
		164 => 'User',
		165 => 'New user',
		166 => 'User edited successfully!',
		167 => 'User not edited successfully!',
		168 => 'User created successfully!',
		169 => 'User not created successfully!',
		170 => 'User deleted successfully!',
		171 => 'User not deleted successfully!',
		172 => 'Username',
		173 => 'eMail',
		174 => 'Actions',
		175 => 'Doubleclick to edit user ',
		176 => ' ',
		177 => 'create eMail',
		178 => 'edit user ',
		179 => ' ',
		180 => 'Edit rights for user ',
		181 => ' ',
		182 => 'Deactivate user ',
		183 => ' ',
		184 => 'Activate user ',
		185 => ' ',
		186 => 'Delete user ',
		187 => ' ',
		188 => 'New user',
		189 => 'Realy delete the selected user?',
		190 => 'No existing user found',
		// Neuer Benutzer
		191 => 'New user - Constructr Administration',
		192 => 'New user',
		193 => 'Username',
		194 => 'eMail',
		195 => 'Password - at least a length of 8 character',
		196 => 'Retype the above password',
		197 => 'Save user',
		198 => 'Cancel',
		199 => 'Press key "ESC" to return to previous view',
		200 => 'Please insert all information in the above user-form!',
		201 => 'Password and password-retype are not the same!',
		// Benutzer editieren
		202 => 'Edit user - Constructr Administration',
		203 => 'Edit user',
		204 => 'Username',
		205 => 'eMail',
		206 => 'Password - at least a length of 8 character - create a new one',
		207 => 'Password retype',
		208 => 'Save user',
		209 => 'Cancel',
		210 => 'Press key "ESC" to return to previous view',
		211 => 'An error occured!',
		212 => 'Please insert all information in the above form!',
		213 => 'Password and password-retype are not the same!',
		// Benutzerrechte editieren
		214 => 'User-rights - Constructr Administration',
		215 => 'User-rights for user',
		216 => ' User-rights',
		217 => 'Cancel',
		218 => 'Right',
		219 => 'Granted',
		220 => 'No',
		221 => 'Yeah',
		222 => 'No user-rights found',
		// Inhalte Übersicht
		223 => 'Content - Constructr Administration',
		224 => 'Content of page',
		225 => 'Open page in new browser-tab',
		226 => ' Content Elements',
		227 => 'New content element',
		228 => 'Content edited successfully!',
		229 => 'Content not edited successfully!',
		230 => 'Content created successfully!',
		231 => 'Content not created successfully',
		232 => 'Content deleted successfully!',
		233 => 'Content not deleted successfully!',
		234 => 'Content moved successfully!',
		235 => 'Content not moved successfully!',
		236 => 'Show preview',
		237 => 'Position',
		238 => 'Content',
		239 => 'Preview',
		240 => 'Actions',
		241 => 'Doubleclick to edit content ',
		242 => '',
		243 => 'Move content',
		244 => 'upwards',
		245 => 'Move content',
		246 => 'downwards',
		247 => 'Create new content before this position ',
		248 => 'Create new content after this position ',
		249 => 'Show content-element',
		250 => '',
		251 => 'Hide content-element',
		252 => '',
		253 => 'Edit content',
		254 => '',
		255 => 'Delete content',
		256 => '',
		257 => 'Content of page',
		258 => 'Show page on new browser tab',
		259 => 'Content-elements',
		260 => 'New content',
		261 => 'Press key "ESC" to return to previous view',
		262 => 'Realy delete this content-element?',
		// Neuer Inhalt / davor / danach
		263 => 'New content-element - Constructr Administration',
		264 => 'New content-element on page ',
		265 => 'Show page in new browser-tab',
		266 => '',
		267 => 'Show uploads',
		268 => '|',
		269 => 'Show pages',
		270 => '',
		271 => 'Please select',
		272 => 'Constructr Mapping',
		273 => 'Content (Markdown)',
		274 => 'Save content',
		275 => 'Cancel',
		276 => 'Press key "ESC" to return to previous view',
		277 => 'This content will be transformed from ',
		278 => 'Markdown',
		279 => 'Markdown-Extra',
		280 => '-syntax into HTML-syntax!',
		281 => 'Existing pages',
		282 => 'Page',
		283 => 'Markdown-Code',
		284 => 'Insert Markdown-Code',
		285 => 'Existing Uploads',
		286 => 'Filename',
		287 => 'Markdown-Code',
		288 => 'Multiple selection',
		289 => 'Insert Markdown-Code',
		290 => 'Insert Markdown-Code',
		291 => 'Insert Pictures',
		292 => 'close',
		293 => 'Chars',
		294 => 'Words',
		295 => 'Paragraphs',
		296 => 'No Preview',
		// Inhalt bearbeiten
		297 => 'Edit Content - Constructr Administration',
		298 => 'Edit content on page ',
		299 => ' ',
		300 => 'Preview in new browser-tab',
		301 => 'edit',
		302 => 'Show uploads',
		303 => '|',
		304 => 'Show pages',
		305 => '',
		306 => 'No Preview',
		307 => 'Please select',
		308 => ' - remove mapping - ',
		309 => 'Constructr Mapping',
		310 => 'Content (Markdown)',
		311 => 'Save content',
		312 => 'Cancel',
		313 => 'Chars',
		314 => 'Words',
		315 => 'Paragraphs',
		316 => 'This content will be transformed from ',
		317 => 'Markdown',
		318 => 'Markdown-Extra',
		319 => '-syntax into HTML-syntax!',
		320 => 'An unknown error occured!',
		321 => 'Existing pages',
		322 => 'Page',
		323 => 'Markdown-Code',
		324 => 'insert Markdown-Code',
		325 => 'close',
		326 => 'Existing uploads',
		327 => 'File',
		328 => 'Markdown-Code',
		329 => 'Multiple selection',
		330 => 'Insert Markdown-Code',
		331 => 'Insert Markdown-Code',
		332 => 'Insert Pictures',
		333 => 'close',
		334 => 'Press key "ESC" to return to previous view',
		// Zwischenablage
		335 => 'Copy Content-Element to clipboard!',
		336 => 'Successfully copied!',
		337 => 'Press Ctrl+C to copy!',
		338 => 'Copy Content-Element!',
		339 => 'Paste Content-Element after this position!',
		340 => 'Error: Not successfully copied!',
		341 => 'Pasted successfully!',
		342 => 'Error: Not successfully pasted!',
		// Suche Uploads
		343 => 'Reset search',
		// Bilder umbenennen
		344 => 'Edit file',
		345 => 'Edit File',
		346 => '',
		347 => 'Error: File not found!',
		348 => 'Cancel',
		349 => 'Filename',
		350 => 'Save File',
		351 => 'Cancel',
		352 => 'Edited successfully!',
		353 => 'Error: Not successfully editet!',
		354 => 'Preview',
		355 => 'Special Image-Effects (Image will be saved as copie)',
		356 => '- Save as inverted image',
		357 => '- Save as grayscale image',
		358 => '- Save as embossed image',
		359 => '- Save as sepia image',
		360 => '- Save as horizontal flipped image',
		361 => '- Save as vertical flipped image',
		362 => '- Save as pixelated image',
		363 => '- Save as rotated (90°) image',
		364 => '- Save as rotated (180°) image',
		365 => '- Save as rotated (270°) image',
		366 => '- Return to original image file and remove effects',
		// MrCleaner
		367 => 'All cached files are deleted successfully (frontend + backend)!',
		368 => 'Clear all cached files',
		// Bildbearbeitung
		369 => 'Height',
		370 => 'Width',
	];
